### PR TITLE
docs(guides/database/full-text-search) update searching multiple columns

### PR DESCRIPTION
**Pull request by k11q**
## What kind of change does this PR introduce?

Update searching multiple columns guide (js and dart) with a workaround involving computed virtual columns

Solution by @steve-chahez from this https://github.com/supabase/postgrest-js/issues/289

Bug fix, feature, docs update, ...

## What is the current behavior?

Please link any relevant issues here.

https://github.com/supabase/supabase/pull/12508
https://github.com/supabase/postgrest-js/issues/289

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: apps/docs/pages/guides/database/full-text-search.mdx [hunk 1]
```diff
@@ -227,6 +227,9 @@ final result = await client
 
 ### Search multiple columns
 
+Right now there is no direct way to use Javascript or Dart to search through multiple columns. A workaround exist where using SQL is required.
```
Comment (gregnr) [inline apps/docs/pages/guides/database/full-text-search.mdx:230]:
```suggestion
Right now there is no direct way to use JavaScript or Dart to search through multiple columns. A workaround exist where using SQL is required.
```

File: apps/docs/pages/guides/database/full-text-search.mdx [hunk 2]
```diff
@@ -247,6 +250,38 @@ where
   @@ to_tsquery('little');
 ```
 
+</TabPanel>
+<TabPanel id="js" label="JavaScript">
+
+```sql
+CREATE FUNCTION title_description(books) RETURNS text AS $$
+  SELECT $1.title || ' ' || $1.description;
+$$ LANGUAGE SQL;
```
Comment (gregnr) [inline apps/docs/pages/guides/database/full-text-search.mdx:259]:
```suggestion
create function title_description(books) returns text as $$
  select $1.title || ' ' || $1.description;
$$ language sql;
```

File: apps/docs/pages/guides/database/full-text-search.mdx [hunk 3]
```diff
@@ -247,6 +250,38 @@ where
...
+  .textSearch('title_description', `little`)
+```
+
+</TabPanel>
+<TabPanel id="dart" label="Dart">
+
+```sql
+CREATE FUNCTION title_description(books) RETURNS text AS $$
+  SELECT $1.title || ' ' || $1.description;
+$$ LANGUAGE SQL;
```
Comment (gregnr) [inline apps/docs/pages/guides/database/full-text-search.mdx:275]:
```suggestion
create function title_description(books) returns text as $$
  select $1.title || ' ' || $1.description;
$$ language sql;
```

**Review by gregnr — CHANGES_REQUESTED** [review]
@khairulhaaziq This is great, thanks for the PR! Just a couple format consistency changes and we'll be good to merge.

File: apps/docs/pages/guides/database/full-text-search.mdx [hunk 4]
```diff
@@ -247,6 +250,38 @@ where
   @@ to_tsquery('little');
 ```
 
+</TabPanel>
+<TabPanel id="js" label="JavaScript">
+
+```sql
+create function title_description(books) returns text as $$
+  select $1.title || ' ' || $1.description;
+$$ language sql;
```
Comment (alaister) [inline apps/docs/pages/guides/database/full-text-search.mdx:259]:
```suggestion
$$ language sql immutable;
```

File: apps/docs/pages/guides/database/full-text-search.mdx [hunk 5]
```diff
@@ -247,6 +250,38 @@ where
...
+  .textSearch('title_description', `little`)
+```
+
+</TabPanel>
+<TabPanel id="dart" label="Dart">
+
+```sql
+create function title_description(books) returns text as $$
+  select $1.title || ' ' || $1.description;
+$$ language sql;
```
Comment (alaister) [inline apps/docs/pages/guides/database/full-text-search.mdx:275]:
```suggestion
$$ language sql immutable;
```

**Review by alaister — COMMENTED** [review]
@khairulhaaziq, Thanks for this PR!

@steve-chavez, should these functions for the computed columns be marked as `immutable`? Or will it not make a difference in this context?

**Comment by khairulhaaziq** [comment]
@steve-chavez @gregnr @alaister I applied all the suggestions. Please let me know if there are things to be adjusted/fixed!

**Review by steve-chavez — APPROVED** [review]
LGTM. Thanks!

**Review by gregnr — APPROVED** [review]

**Pull request merged**